lexoid/core/parse_type/static_parser.py

```diff
@@ -47,8 +58,11 @@ def parse_static_doc(path: str, **kwargs) -> Dict:
                 "parent_title": kwargs.get("parent_title", ""),
                 "recursive_docs": [],
             }
-    elif file_type == "text/csv":
-        df = pd.read_csv(path)
+    elif file_type == "text/csv" or "spreadsheet" in file_type:
+        if "spreadsheet" in file_type:
+            df = pd.read_excel(path)
```

Comment:
Consider setting `chunksize` to not blow memory? Maybe `chunksize=10000`, also `memory_map=True`?
For this iteration, maybe ok with a simple loop?

Later we cab evolve to a better data structure?

Reply:
Sorry, didn't get what you meant. We could divide the rows into segments, but I'm not sure how that helps with the memory.

Reply:
Supporting large `.csv` will bring in additional complexities which maybe we address in future.
So, was suggesting even for smaller/medium size `.csv` we do upload in chunks to control performance?

Reply:
But even if we read in chunks, all the rows will eventually be in memory (in the return dictionary as Markdown), right? I suppose there could be a temporary overhead memory usage for each read, and that might be less for fewer rows, but I'm wondering if that improvement is worth the increase in read time.

Reply:
> I suppose there could be a temporary overhead memory usage for each read, and that might be less for fewer rows, but I'm wondering if that improvement is worth the increase in read time.

Will help in avoiding memory crashes n slowness for large files in-case that happens. 
Even in limited form of chunking strategies performance could further be improved using concurrent processing or other frameworks like dask/ etc. But, thinking we can look into that later because it might start to bring in additional complexities? https://motherduck.com/blog/duckdb-versus-pandas-versus-polars/